src/lib/metrics.js

```diff
@@ -141,6 +142,14 @@ const prometheus = (reqType, dataSet, username, client, groups, duration) => {
     help: `Request time of requests from members of the ${g} group`
   }));
 
+  if(blocked) {
+    prometheusMetrics.req[dataSet].blocked = prometheusMetrics.req[dataSet].blocked || new promClient.Counter({
+      name: `${prometheusPrefix}_req_${escape(dataSet)}_blocked`,
+      help: `Number of blocked records for ${dataSet}`
+    });
```

Comment:
I think this should be taken outside of the if statement.

Reply:
@daniel-ac-martin is there a reason this setup happens at request time and not as a one-off at application start-up?

Reply:
Probably!

Oh I think it's because we don't have all the information we need at start-up time.

e.g. We don't know all of the groups.

I agree it's a bit skanky though.

Now that we've largely given up on most of these metrics, I wonder if we could simplify this code so that we could instantiate on start-up...

Reply:
We could certainly do most of the init at startup and just check groups at request time.